### ISSUE TITLE
Download link not working for encrypted rooms in Element Web

### ISSUE THREAD

**Issue by ornxka** [vector-im/element-web#18315]
When I click on the download button in Element in an encrypted room, nothing happens. I can see in the browser network log it downloads the file the first time the button is clicked, but it doesn't open a file save dialog. However, downloading files works for files that have previews, e.g. JPG, but not any other files.

Versions:
* Element Web version 1.7.34-rc.1
* nginx reverse proxy
* matrix-synapse 1.34.0 (my homeserver, not eg matrix.org)
* olm version 3.2.3
* Chromium 91.0.4472.114 (Official Build) (64-bit)
* Also tried on Firefox and got the same thing except the spinner spun indefinitely after I clicked download, instead of spinning for a second and then returning to the normal download icon

**Comment by turt2live** [comment]
@ornxka what sort of file was it that you were trying to download?

**Comment by ornxka** [comment]
This behavior seems to occur for all files that can't be previewed by Element, but in particular I've seen it with .pdf and .torrent files.

**Comment by turt2live** [comment]
are you able to send over logs for this issue? After triggering the problem, go to Settings -> Help -> Submit Debug Logs and put this issue URL into the box.

**Comment by ornxka** [comment]
I have found the problem - it was that I had set the CSP to `frame-ancestors: 'none'`. This was preventing the decrypted response from being created in an iframe and initiating the download dialog, and I was getting an ERR_BLOCKED_BY_RESPONSE error which was not displaying in the UI. I solved it by setting it to `frame-ancestors 'self'`, and now downloading files works again. I'm not sure if there is a better solution, since I would rather have left it at "none" as the README in https://github.com/vector-im/element-web says to do.
